In [1]:
import pandas as pd
import numpy as np
import glob

path =r'C:\WorkData\Avito\Avito_Daily' # use your path
allFiles = glob.glob(path + "/*.csv")
frame = pd.DataFrame()
datalist_ = [] #list with all data
for file_ in allFiles:
    print('opening', file_)
    df = pd.read_csv(file_,index_col=None, names=['Name', 'Price', 'Date', 'Link'], header=0)
    #df.drop_duplicates(subset='Link', keep="last", inplace = True)
    datalist_.append(df)
print(str(len(datalist_))+' files opened')
GPUframe = pd.concat(datalist_, axis=0, join='outer', join_axes=None, ignore_index=False,
          keys=None, levels=None, names=None, verify_integrity=False,
          copy=True)

#GPUframe['Price'] = GPUframe['Price']
GPUframe = GPUframe.set_index('Date')
GPUframe['Price'] = GPUframe.apply(lambda x: x[1][:-3], axis=1)

opening C:\WorkData\Avito\Avito_Daily\GPU_04_25_2018.csv
opening C:\WorkData\Avito\Avito_Daily\GPU_04_26_2018.csv
opening C:\WorkData\Avito\Avito_Daily\GPU_04_27_2018.csv
opening C:\WorkData\Avito\Avito_Daily\GPU_04_28_2018.csv
opening C:\WorkData\Avito\Avito_Daily\GPU_04_29_2018.csv
opening C:\WorkData\Avito\Avito_Daily\GPU_04_30_2018.csv
opening C:\WorkData\Avito\Avito_Daily\GPU_05_01_2018.csv
opening C:\WorkData\Avito\Avito_Daily\GPU_05_02_2018.csv
opening C:\WorkData\Avito\Avito_Daily\GPU_05_03_2018.csv
opening C:\WorkData\Avito\Avito_Daily\GPU_05_04_2018.csv
opening C:\WorkData\Avito\Avito_Daily\GPU_05_05_2018.csv
opening C:\WorkData\Avito\Avito_Daily\GPU_05_06_2018.csv
opening C:\WorkData\Avito\Avito_Daily\GPU_05_07_2018.csv
opening C:\WorkData\Avito\Avito_Daily\GPU_05_08_2018.csv
opening C:\WorkData\Avito\Avito_Daily\GPU_05_09_2018.csv
opening C:\WorkData\Avito\Avito_Daily\GPU_05_10_2018.csv
opening C:\WorkData\Avito\Avito_Daily\GPU_05_11_2018.csv
opening C:\WorkData\Avito\Avito

In [2]:
GPUframe['Name'] = GPUframe['Name'].str.lower()

In [3]:
GPUframe.Name = GPUframe.Name.str.replace('gtx', 'GTX ')
GPUframe.Name = GPUframe.Name.str.replace('rx', 'RX ')

In [4]:
GPUframe['Price'] =  GPUframe['Price'].str.replace('\s+','')
GPUframe = GPUframe[GPUframe['Price'].apply(lambda x: str(x).isdigit())]

In [5]:
GPUframe['Price'].astype(int)

Date
04_25_2018     13000
04_25_2018     99000
04_25_2018     23999
04_25_2018    120000
04_25_2018     10000
04_25_2018       500
04_25_2018       500
04_25_2018      1000
04_25_2018     12500
04_25_2018      1500
04_25_2018      3500
04_25_2018     49000
04_25_2018     90000
04_25_2018     23300
04_25_2018     75000
04_25_2018       200
04_25_2018     38500
04_25_2018    180000
04_25_2018      3000
04_25_2018    125000
04_25_2018     19500
04_25_2018      9000
04_25_2018     42000
04_25_2018     90000
04_25_2018     20000
04_25_2018       500
04_25_2018       350
04_25_2018    130000
04_25_2018     25000
04_25_2018     18000
               ...  
06_15_2018      1000
06_15_2018     14000
06_15_2018     27000
06_15_2018      6000
06_15_2018     29000
06_15_2018     17900
06_15_2018       800
06_15_2018      2500
06_15_2018      1500
06_15_2018     29990
06_15_2018      7500
06_15_2018       300
06_15_2018     35490
06_15_2018      4000
06_15_2018      2000
06_15_2018     46000
06_15_20

In [6]:
GPUnames = ['GTX', 'gtx', 'RX', 'rx']
GPUfiltered = GPUframe[GPUframe.Name.str.contains('|'.join(GPUnames))]
GPUnotFarm = GPUfiltered[GPUfiltered.Name.str.contains('ферма') == False]

In [7]:
splittedName = GPUnotFarm.Name.str.split()

In [8]:
GPUnotFarm

Name   Price  \
Date                                                                    
04_25_2018          видеокарта asus geforce GTX  1050ti strix   13000   
04_25_2018    sapphire RX  570 8 gb radeon nitro+. видеокарта   23999   
04_25_2018   видео карты gigabyte pci-e gv-np106d59-6g GTX...  120000   
04_25_2018                          gigabyte geforce GTX  780   10000   
04_25_2018                                      GTX  970 evga   12500   
04_25_2018                               видеокарта 4GTX  660    3500   
04_25_2018                                       GTX  1080 ti   49000   
04_25_2018                GTX  1060 rog strix ddr5 samsung 6g   23300   
04_25_2018                         gigabyte aorus RX  580 4gb   75000   
04_25_2018                     видеокарта geforce GTX  1070ti   38500   
04_25_2018       видеокарта asus geforce GTX  780 на запчасти    3000   
04_25_2018                       видеокарта radeon RX  560 4g    9000   
04_25_2018                                      msi GTX  1080   42000   
04_25_2018      видеокарта radeon RX  570 4 g asus expedition   20000   
04_25_2018                 asus rog strix RX 580 8 gb samsung   25000   
04_25_2018                 видеокарты gigabyte GTX  1060, 3gb   18000   
04_25_2018         gigabyte GTX  1060 g1 gaming 6gb (rev.2.0)   23000   
04_25_2018                                    ga GTX  650 2gb    2500   
04_25_2018            нерабочая gigabyte geforce GTX  1080 ti   17000   
04_25_2018         видеокарта nvidia geforce GTX  750 ti 2 гб    6000   
04_25_2018                            zotac GTX 760 2gb gddr5    4000   
04_25_2018                                     пк GTX  1080ti  110000   
04_25_2018   gigabyte radeon RX  580 gaming 8192mb gddr5 н...   24000   
04_25_2018                  видеокарта GTX  1070 windforce oc   31990   
04_25_2018          asus geforce GTX  1050 ti 4096mb cerberus   13990   
04_25_2018    asus geforce GTX  1060 dual-GTX 1060-6g 6144 mb   28700   
04_25_2018      asus geforce GTX  1050ti oc ex-GTX 1050ti-o4g   14700   
04_25_2018              msi geforce GTX  1080 ti gaming x 11g   53000   
04_25_2018                                RX  560 asus 4g evo    9500   
04_25_2018               sapphire radeon RX 570 8gb на 8 карт  290000   
...                                                       ...     ...   
06_15_2018          видеокарта palit geforce GTX  1050 ti 4gb    8000   
06_15_2018                 видеокарта palit geforce GTX  1070   24500   
06_15_2018                     gigabyte RX 580 8gb, (RX  580)   19600   
06_15_2018                       palit nvidia GTX 1060 6144mb   13000   
06_15_2018         видеокарта amd sapphire RX  580 nitro+ 4gb   20000   
06_15_2018   gigabyte RX 580 gaming 8g mi(gv-RX 580gaming-...   19800   
06_15_2018               видеокарта v560n-1024b. GTX 560 ddr5    1500   
06_15_2018        видеокарта asus geforce GTX  1060 6gb gddr5   19500   
06_15_2018                     asus geforce 9800 GTX + 738mhz    3000   
06_15_2018                             asus strix GTX  1050ti   12000   
06_15_2018   видеокарта palit GTX  1070 ti super jetstream...   36850   
06_15_2018   видеокарта sapphire amd radeon RX  550 pulse 4gb    7000   
06_15_2018   antminer s9 13,5 ths sn: it06yt RX -6948 в на...   54000   
06_15_2018                          geforce GTX  560 ti palit    2500   
06_15_2018             продам palit pa-GTX 1060 dual 6g gddr5   18000   
06_15_2018                      GTX  960 gaming windforce 4gb   13000   
06_15_2018                   видеокарты rog strix RX  570 4gb   17000   
06_15_2018        видеокарта gigabyte GTX  970 g1 gaming 4 gb   14000   
06_15_2018               palit geforce GTX  1070 ti. в идеале   27000   
06_15_2018                       nvidia GTX  750 ti 2 gb asus    6000   
06_15_2018        видеокарта sapphire pulse radeon RX  580 8g   17900   
06_15_2018                                       GTX  285 1gb    1500   
06_15_2018                            rade

## Regex combination 1

In [9]:


#103 unique

import re

ShortNameList = []
Exceptions = []

NvidiaPattern = r'([G*][T*][X*])(\s*)(\d*)(\s*)(t?i?)'
#AMDPattern = r'([R*][X*](\s*)(\d*))'
AMDPattern = r'([R*][X*](\s*)(\d*)(v?e?g?a?)(\s*)(\d{2}))'

for name in GPUnotFarm['Name']:
    if len(re.findall(NvidiaPattern, name)) != 0:
        ShortNameList.append(re.search(NvidiaPattern, name).group(1)+' '+re.search(NvidiaPattern, name).group(3)+re.search(NvidiaPattern, name).group(5))
    elif len(re.findall(AMDPattern, name)) != 0:
        ShortNameList.append(re.search(AMDPattern, name).group(0))
    else:
        ShortNameList.append('Not what I need')
        Exceptions.append(name)# regex exceptions
print(ShortNameList, len(ShortNameList))

['GTX 1050ti', 'RX  570', 'GTX 1060', 'GTX 780', 'GTX 970', 'GTX 660', 'GTX 1080ti', 'GTX 1060', 'RX  580', 'GTX 1070ti', 'GTX 780', 'RX  560', 'GTX 1080', 'RX  570', 'RX 580', 'GTX 1060', 'GTX 1060', 'GTX 650', 'GTX 1080ti', 'GTX 750ti', 'GTX 760', 'GTX 1080ti', 'RX  580', 'GTX 1070', 'GTX 1050ti', 'GTX 1060', 'GTX 1050ti', 'GTX 1080ti', 'RX  560', 'RX 570', 'RX 570', 'GTX 1080', 'GTX 1070', 'GTX 1060', 'GTX 1070ti', 'RX 570', 'GTX 1070', 'GTX 1060', 'RX  570', 'RX  570', 'GTX 970', 'GTX 1070', 'GTX 1070ti', 'RX  580', 'GTX 1070', 'GTX 650ti', 'RX  570', 'GTX 970', 'GTX 1070', 'GTX 660', 'GTX 1070', 'GTX 1060', 'GTX 460', 'RX 580', 'GTX 1080ti', 'GTX 1070', 'GTX 1070ti', 'RX  580', 'GTX 1070', 'GTX 650ti', 'GTX 745', 'RX 460', 'GTX 1060', 'GTX 1080ti', 'GTX 1050ti', 'GTX 960', 'RX  470', 'GTX 1060', 'GTX 680', 'GTX 1080', 'RX  580', 'GTX 1070', 'GTX 1070', 'GTX 560', 'GTX 760', 'GTX 1050ti', 'GTX 1050ti', 'GTX 1070', 'GTX 1060', 'GTX 1070', 'RX  580', 'GTX 1070', 'GTX 1060', 'GTX 1070

## Regex combination 2
#### 1st try = 103 unique
- NvidiaPattern = r'([G*][T*][X*])(\s*)(\d*)(\s*)(t?i?)'
- AMDPattern = r'([R*][X*](\s*)(\d*)(v?e?g?a?)(\s*)(\d{2}))'

#### 2nd try = 88 unique
improving NvidiaPattern
- NvidiaPattern = r'([G*][T*][X*])(\s*)(\d*)(\s*)(t{1}i{1})*'
improving AMDPattern
- AMDPattern = r'([R*][X*](\s*)(\d*)((v{1}e{1}g{1}a{1})?)(\s*)(\d{2-3})(\D))'

In [10]:
import re

ShortNameList = []
Exceptions = []

NvidiaPattern = r'([G*][T*][X*])(\s*)(\d*)(\s*)(t{1}i{1})*'
AMDPattern = r'([R*][X*](\s*)(\d{3}))'
AMDVegaPattern = r'([R*][X*](\s*)(vega)(\s*)(\d*))'


for name in GPUnotFarm['Name']:
    if len(re.findall(NvidiaPattern, name)) != 0:
        ShortNameList.append(re.search(NvidiaPattern, name).group(0))
    elif len(re.findall(AMDPattern, name)) != 0:
        ShortNameList.append(re.search(AMDPattern, name).group(0))
    elif len(re.findall(AMDVegaPattern, name)) != 0:
        ShortNameList.append(re.search(AMDVegaPattern, name).group(0))    
    else:
        ShortNameList.append('Not what I need')
        Exceptions.append(name)# regex exceptions
print(ShortNameList, len(ShortNameList))

['GTX  1050ti', 'RX  570', 'GTX  1060', 'GTX  780', 'GTX  970 ', 'GTX  660', 'GTX  1080 ti', 'GTX  1060 ', 'RX  580', 'GTX  1070ti', 'GTX  780 ', 'RX  560', 'GTX  1080', 'RX  570', 'RX 580', 'GTX  1060', 'GTX  1060 ', 'GTX  650 ', 'GTX  1080 ti', 'GTX  750 ti', 'GTX 760 ', 'GTX  1080ti', 'RX  580', 'GTX  1070 ', 'GTX  1050 ti', 'GTX  1060 ', 'GTX  1050ti', 'GTX  1080 ti', 'RX  560', 'RX 570', 'RX 570', 'GTX 1080 ', 'GTX  1070 ', 'GTX  1060 ', 'GTX  1070 ti', 'RX 570', 'GTX 1070 ', 'GTX  1060 ', 'RX  570', 'RX  570', 'GTX  970 ', 'GTX  1070 ', 'GTX  1070 ti', 'RX  580', 'GTX  1070 ', 'GTX  650 ti', 'RX  570', 'GTX  970 ', 'GTX  1070 ', 'GTX 660 ', 'GTX  1070 ', 'GTX  1060 ', 'GTX  460 ', 'RX 580', 'GTX  1080ti', 'GTX  1070 ', 'GTX  1070 ti', 'RX  580', 'GTX  1070 ', 'GTX  650 ti', 'GTX  745 ', 'RX 460', 'GTX 1060 ', 'GTX  1080 ti', 'GTX  1050ti', 'GTX  960 ', 'RX  470', 'GTX 1060 ', 'GTX  680 ', 'GTX  1080 ', 'RX  580', 'GTX  1070 ', 'GTX 1070 ', 'GTX 560 ', 'GTX 760 ', 'GTX  1050ti', '

In [11]:
Exceptions, len(Exceptions)

([' antminer s9 13,5 ths sn: xv05rв RX -4799 в наличии',
  ' видеокарта sapphire radeon RX -460 2gb',
  ' видеокарта gigabyte RX -570 4gb',
  ' оперативная память hypeRX  ddr3(8gb)',
  ' antminer s9 13,5 ths sn: cl06yt RX -0053 в наличии',
  ' antminer s9 13,5 ths sn: it06yt RX -6948 в наличии'],
 6)

In [12]:
#Dealing with whitespaces

ShortNameListFixed = []

for a in ShortNameList:
    a = ''.join(a.split())
    ShortNameListFixed.append(a)
    
print(ShortNameListFixed,len(ShortNameListFixed))

['GTX1050ti', 'RX570', 'GTX1060', 'GTX780', 'GTX970', 'GTX660', 'GTX1080ti', 'GTX1060', 'RX580', 'GTX1070ti', 'GTX780', 'RX560', 'GTX1080', 'RX570', 'RX580', 'GTX1060', 'GTX1060', 'GTX650', 'GTX1080ti', 'GTX750ti', 'GTX760', 'GTX1080ti', 'RX580', 'GTX1070', 'GTX1050ti', 'GTX1060', 'GTX1050ti', 'GTX1080ti', 'RX560', 'RX570', 'RX570', 'GTX1080', 'GTX1070', 'GTX1060', 'GTX1070ti', 'RX570', 'GTX1070', 'GTX1060', 'RX570', 'RX570', 'GTX970', 'GTX1070', 'GTX1070ti', 'RX580', 'GTX1070', 'GTX650ti', 'RX570', 'GTX970', 'GTX1070', 'GTX660', 'GTX1070', 'GTX1060', 'GTX460', 'RX580', 'GTX1080ti', 'GTX1070', 'GTX1070ti', 'RX580', 'GTX1070', 'GTX650ti', 'GTX745', 'RX460', 'GTX1060', 'GTX1080ti', 'GTX1050ti', 'GTX960', 'RX470', 'GTX1060', 'GTX680', 'GTX1080', 'RX580', 'GTX1070', 'GTX1070', 'GTX560', 'GTX760', 'GTX1050ti', 'GTX1050ti', 'GTX1070', 'GTX1060', 'GTX1070', 'RX580', 'GTX1070', 'GTX1060', 'GTX1070', 'GTX1060', 'RX580', 'GTX1050ti', 'GTX2', 'GTX1060', 'GTX760', 'GTX1060', 'GTX1080ti', 'GTX650',

In [13]:
GPUnotFarm.loc[:,'ShortName'] = ShortNameListFixed

C:\Anaconda3\lib\site-packages\pandas\core\indexing.py:362: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self.obj[key] = _infer_fill_value(value)
C:\Anaconda3\lib\site-packages\pandas\core\indexing.py:543: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self.obj[item] = s


In [14]:
print(len(GPUnotFarm.ShortName.unique()))
GPUnotFarm.ShortName.unique()

78


array(['GTX1050ti', 'RX570', 'GTX1060', 'GTX780', 'GTX970', 'GTX660',
       'GTX1080ti', 'RX580', 'GTX1070ti', 'RX560', 'GTX1080', 'GTX650',
       'GTX750ti', 'GTX760', 'GTX1070', 'GTX650ti', 'GTX460', 'GTX745',
       'RX460', 'GTX960', 'RX470', 'GTX680', 'GTX560', 'GTX2', 'GTX560ti',
       'GTX770', 'RX550', 'GTX295', 'GTX630', 'GTX980', 'GTX', 'RX500',
       'RXvega64', 'GTX550ti', 'GTX6', 'GTX260', 'GTX950', 'GTX750',
       'GTX670', 'GTX580', 'RX480', 'GTX780ti', 'GTX1050', 'GTX570',
       'GTX285', 'GTX480', 'GTXti', 'GTX660ti', 'GTX1', 'GTX690', 'RX155',
       'GTX470', 'GTX420', 'GTX980ti', 'GTX4', 'GTX590', 'GTX275',
       'GTX250', 'GTX9600', 'GTX675', 'GTX575', 'RXvega56', 'RX960',
       'GTX860', 'GTX465', 'GTX1015ti', 'GTX10', 'GTX880', 'NotwhatIneed',
       'GTX8800', 'GTX450', 'GTX730', 'GTX512', 'GTX8393ti', 'GTX280',
       'GTX440', 'GTX8', 'GTX1607'], dtype=object)

### Dealing with irrelevant unique GPU

- видеокарта для ноутбука nvidia 680m GTX 2gb == nvidia 680m GTX 
- GTX 6 == 1060 GTX 6gb need to use regex without dependence on str position in text
- RX 570 12 ==  RX 570 1268mhz
- RX vega 64 and RX vega64
- GTX ti == gtx titan and mismatch
- GTX 660t == GTX 660
- GTX 1 == mismatch, bad name
- RX 155256 == mismatch, bad name
- GTX 1070t == GTX 1070 turbo
- RX 460 12 == radeon RX 460 1212mhz
- RX 460 40 == RX 460 4096mb
- RX 560 11 == RX 560 1196mhz
- GTX t == n570GTX twin frozr iii 
- RX 460 20 == RX 460 2048mb
- GTX 1015ti == dolbaeb
- GTX 10 == dolbaeb
- GTX 560t == geforce GTX 560 twin frozr ii

In [15]:
GPUnotFarm.loc[GPUnotFarm['ShortName'] == 'GTX']

Name  Price  \
Date                                                                   
04_26_2018                                asus strix 980 GTX   15000   
04_27_2018                asus geforce (en9800GTX ) (8800gts)    500   
04_29_2018     видеокарта, nvidia geforce GTX  n750ti 2gd5/oc   6000   
04_30_2018   видеокарта xfx radeon RX  580 GTX  xxx editio...  29000   
04_30_2018         1080ti,1080,1070, 1060 6gb GTX  видеокарты  50000   
04_30_2018               видеокарта msi GTX  geforce 1050 2gb  10000   
04_30_2018                         видеокарта nvidia 8800GTX     800   
05_02_2018               видеокарта msi GTX  geforce 1050 2gb   8500   
05_03_2018                      g750jh mxm n14e GTX  rev. 2.1   2000   
05_04_2018                          видеокарта 7800GTX  pci-e    600   
05_04_2018    видеокарта gigabyte geforce GTX  g1 gaming 1060  24000   
05_04_2018                          видеокарта nvidia 560GTX    2000   
05_04_2018                        msi 1060 GTX  gaming x 6 gb  21500   
05_05_2018                             geforce 460 GTX  palit   2790   
05_06_2018   видеокарта msi h680GTX  twin frozr 4g ddr5/oc...  10000   
05_06_2018           вдяное охлаждение для видеокарты 680GTX    3500   
05_06_2018                       msi n460GTX  cyclone 1gd5/oc   2300   
05_07_2018                           n9800GTX  nvidia geforce   3000   
05_08_2018   geforce GTX  sli bridge, sli мост nvidia 3 слота   2000   
05_08_2018        топовая из 5й серии GTX -580 3gb видеокарта   4800   
05_08_2018                          msi n550GTX -ti cuclone 2   2000   
05_08_2018   видеокарта ноутбука mxm nvidia geforce 570m GTX    8000   
05_09_2018                видеокарта n9800GTX -512mx-f-89-206    500   
05_09_2018                nvidia geforce 9800 GTX /9800 GTX +   1500   
05_09_2018   asus gf 9800 GTX + 738mhz pci-e 2.0 512mb 256...   1100   
05_11_2018               видеокарта nvidia geforce nx7800GTX     500   
05_11_2018                     msi geforce GTX  gaming x 1080  39999   
05_12_2018            видеокарты msi GTX  gaming x 1070, 1060  31000   
05_12_2018   видеокарта nvidia 765m GTX  для samsung np700g7c  14500   
05_13_2018                            видеокарта msi n260GTX    1500   
...                                                       ...    ...   
05_22_2018                            zotac geforce 9800 GTX     500   
05_22_2018       asus geforce GTX  (strix-GTX 1070-8g-gaming)  38300   
05_22_2018                             видеокарта msi-460GTX    1800   
05_23_2018   видеокарта msi n570GTX  twin frozr iii powere...   4500   
05_23_2018      видеокарта 9800m GTX  для ноутбука dell m1730   3000   
05_24_2018                            msi GTX  geforce 550 ti   4000   
05_24_2018                 видеокарта nvidia geforce 980 GTX   15000   
05_29_2018                        видеокарта nvidia n460 GTX     800   
05_29_2018                        msi 1080 GTX  gaming x 8 gb  39000   
05_30_2018                          продам msi 560GTX -se 1gb   2000   
05_31_2018                                   cyclone n460GTX    1400   
06_01_2018   видеокарта msi n570GTX  twin frozr iii oc нер...   1000   
06_02_2018                      g750jh mxm n14e GTX  rev. 2.1   2000   
06_03_2018                         видеокарта nvidia 8800GTX     800   
06_05_2018                       msi n460GTX  cyclone 1gd5/oc   2000   
06_05_2018   каркасы(риг) для майнинг ферм RX  GTX  radeon...    450   
06_05_2018                             geforce 460 GTX  palit   2790   
06_06_2018                           n9800GTX  nvidia geforce   3000   
06_07_2018   видеокарта ноутбука mxm nvidia geforce 570m GTX    8000   
06_08_2018                  asus GTX  dual 1060 3gb в наличии  18000   
06_09_2018                nvidia geforce 9800 GTX /9800 GTX +   1500   
06_09_2018                        msi 1060 GTX  gaming x 6 gb  20000   
06_10_2018                            видеокарта geforce GTX   29500   
06_11_2018               

In [16]:
#setting price lower price limit and lower bound for gpu model occurences
ModelCounter = GPUnotFarm.ShortName.value_counts()
ModelCounter = ModelCounter.where(ModelCounter >= 30) # Setting threshold 
ModelCounter = ModelCounter.dropna()
ModelCounter, len(ModelCounter)

(GTX1060      664.0
 GTX1070      314.0
 RX580        282.0
 GTX1080ti    264.0
 RX570        229.0
 GTX1080      211.0
 GTX1050ti    194.0
 GTX1070ti    148.0
 RX470        108.0
 GTX760        88.0
 GTX750ti      71.0
 GTX1050       69.0
 GTX           68.0
 GTX970        67.0
 GTX650        62.0
 GTX960        58.0
 GTX660        54.0
 GTX770        52.0
 GTX560ti      50.0
 GTX550ti      48.0
 GTX950        48.0
 GTX560        47.0
 RX460         44.0
 RX560         41.0
 GTX780        40.0
 GTX650ti      35.0
 RX480         34.0
 GTX980        32.0
 GTX750        31.0
 GTX460        30.0
 GTX570        30.0
 RX550         30.0
 Name: ShortName, dtype: float64, 32)

In [17]:
GPUnotFarm['Price'] = pd.to_numeric(GPUnotFarm['Price'])
GPUnotFarm = GPUnotFarm[GPUnotFarm['ShortName'].isin(ModelCounter.index)]
GPUnotFarm = GPUnotFarm[GPUnotFarm.ShortName != 'GTX']
GPUnotFarm = GPUnotFarm.reset_index()

C:\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [18]:
GPUnotFarmMean = GPUnotFarm.groupby('ShortName', as_index=False).mean()

In [19]:
GPUnotFarmMean.rename(columns={"Price": "PriceAvg"}, inplace = True)
GPUnotFarmMean.set_index(['ShortName'],inplace = True)
GPUnotFarmMean

PriceAvg
ShortName               
GTX1050      7656.072464
GTX1050ti   12510.365979
GTX1060     19908.771084
GTX1070     36541.808917
GTX1070ti   36477.087838
GTX1080     40176.900474
GTX1080ti  110477.378788
GTX460       2089.966667
GTX550ti     2379.125000
GTX560       2226.553191
GTX560ti     2575.760000
GTX570       3293.333333
GTX650       2678.048387
GTX650ti     3428.542857
GTX660       4820.370370
GTX750       4624.419355
GTX750ti     5513.225352
GTX760       7311.147727
GTX770       7832.653846
GTX780       9034.700000
GTX950       7114.854167
GTX960       8855.982759
GTX970      13399.746269
GTX980      17987.187500
RX460        7602.204545
RX470       28900.861111
RX480       23896.441176
RX550        6914.266667
RX560        8155.804878
RX570       24215.174672
RX580       27923.358156

In [20]:
GPUnotFarmMean.dtypes

PriceAvg    float64
dtype: object

In [21]:
GPUnotFarm.dtypes

Date         object
Name         object
Price         int64
Link         object
ShortName    object
dtype: object

In [22]:
#test = pd.concat([GPUnotFarm, GPUnotFarmMean])
#test

GPUnotFarmFull = pd.merge(GPUnotFarm, GPUnotFarmMean, left_on='ShortName',right_index=True, how='left', sort=False);
GPUnotFarmFull.sort_values(by=['PriceAvg'], inplace = True)
#GPUnotFarmFull['Date'] = dates
#pd.set_option('display.width', 80) #default 80
GPUnotFarmFull

Date                                               Name    Price  \
777   05_06_2018    видеокарта gigabyte geforce GTX  460 se 1024 мб     1800   
1237  05_14_2018                                  msi GTX  460 hawk      500   
1793  05_23_2018       gainward geforce GTX  460 800mhz pci-e 768mb     1300   
2715  06_04_2018   asus geforce GTX  460 enGTX 460 directcu/2di/...     5000   
719   05_05_2018   asus geforce GTX  460 enGTX 460 directcu/2di/...     5000   
2711  06_04_2018                     видеокарта msi nvidia GTX  460     1600   
2202  05_28_2018                          gigabyte geforce GTX  460     1800   
1235  05_14_2018        zotac GTX  460 synergy 768mb gddr5 (192bit)     1500   
3132  06_09_2018                                gigabyte GTX  460se     2000   
2219  05_29_2018                    palit GTX 460 768m gddr5 192bit     1800   
1480  05_18_2018                              GTX  460 hawk edition     1500   
1026  05_10_2018                                 geforce GTX 460 1g     2000   
133   04_26_2018                      zotac geforce GTX  460 1024mb     1900   
52    04_25_2018          GTX  460 sonic 1024mb gddr5 256 bit palit      999   
1500  05_19_2018              geforce GTX  460v2 (gf114 GTX 560/ti)     2000   
1499  05_19_2018                           видеокарта asus GTX  460     2000   
376   04_30_2018                        видеокарта msi GTX  460 1gb     3900   
2660  06_03_2018                        nvidia GTX  460 1gb 256 bit     1500   
3290  06_12_2018             видеокарта nvidia geforce GTX 460 asus     4000   
1027  05_10_2018                                           GTX  460     2500   
187   04_27_2018                                       GTX  460 1gb     2000   
3406  06_14_2018              видеокарта GTX 460 sonic platinum 1gb     2000   
818   05_06_2018           gainward geforce GTX  460 800mhz pci-e 2     1500   
960   05_08_2018       palit geforce GTX 460 768gb pci-e видеокарта     1800   
2386  05_31_2018                        видеокарта msi GTX  460 1gb     3500   
3002  06_08_2018                            nvidia geforce GTX  460     1300   
2438  06_01_2018                asus en GTX  460 top/2di/256bit/1gb     3000   
1897  05_25_2018             видеокарта GTX 460 smart edition 1024m      500   
1104  05_11_2018   видеокарта pci-e palit GTX 460 1gb gddr5 256 bit     2000   
1355  05_17_2018   видеокарта geforce GTX 460 v2 gainward (нераб...      500   
...          ...                                                ...      ...   
2046  05_26_2018                             evga GTX  1080 ti ftw3    65000   
2044  05_26_2018      видеокарта asus geforce GTX  1080 ti strix oc    62000   
2946  06_07_2018     видеокарта asus geforce GTX  1080 ti rog -o11g    69800   
3200  06_10_2018               gigabyte geforce GTX  1080 ti xtreme   580000   
3199  06_10_2018       видеокарта asus rog strix GTX  1080 ti o11gb    50000   
1583  05_20_2018   asus geforce GTX  1080 ti strix o11g-oc распр...    49900   
1584  05_20_2018                       msi GTX  1080 ti aero 11g oc    45000   
1579  05_20_2018             asus rog geforce GTX  1080 ti strix oc    51000   
557   05_03_2018               gigabyte geforce GTX  1080 ti xtreme  1040000   
556   05_03_2018         gigabyte aorus GTX  1080 ti xtreme edition    52000   
1546  05_20_2018                                geforce GTX  1080ti    55000   
2132  05_28_2018                evga GTX  1080 ti ftw3 hydro copper    65000   
246   04_28_2018                evga GTX  1080 ti ftw3 hydro copper    76000   
2130  05_28_2018        zotac GTX  1080 ti amp extreme core edition    65000   
1519  05_19_2018     palit geforce GTX  1080ti super jetstream 11gb    60700   
2890  06_06_2018     asus geforce GTX  1080 ti rog-strix-GTX 1080ti    62000   
245   04_28_2018       aorus GTX  1080 ti xtreme edition waterforce    75000   
545   05_03_2018                           msi GTX 1080 ti gaming x    71000   
816   05_06_

In [23]:
b = GPUnotFarmFull.loc[557,'Link'] #change index
b = 'https://www.avito.ru' + str(b)
print(b)

https://www.avito.ru/sankt-peterburg/tovary_dlya_kompyutera/gigabyte_geforce_gtx_1080_ti_xtreme_1666119947


In [24]:
from datetime import datetime
olddates = GPUnotFarm['Date'].tolist()
dates = []
for mdy in olddates:
    dates.append(datetime.strptime(mdy, '%m_%d_%Y'))
GPUnotFarmFull['Date'] = dates
GPUnotFarm['Date'] = dates

In [25]:
GPUnotFarm.dtypes

Date         datetime64[ns]
Name                 object
Price                 int64
Link                 object
ShortName            object
dtype: object

In [26]:
GPUnotFarmFull.dtypes

Date         datetime64[ns]
Name                 object
Price                 int64
Link                 object
ShortName            object
PriceAvg            float64
dtype: object

## Searching GPU

In [30]:
GPUnotFarmSearch = GPUnotFarm
GPUnotFarmSearch['Date'] = pd.to_datetime(GPUnotFarm.Date)
GPUnotFarmSearch = GPUnotFarm[GPUnotFarm.ShortName == 'GTX1050']
GPUnotFarmSearch = GPUnotFarmSearch.groupby(['Date','Price'], as_index=False).min()
GPUnotFarmSearch.sort_values(['Date'], ascending=True)
GPUnotFarmSearch.tail(25)

Date  Price                                               Name  \
41 2018-05-25   2400                         видеокарта GTX  1050 китай   
42 2018-05-25   8000            gigabyte geforce GTX  1050 2048 mb gdd5   
43 2018-05-25   8700                        видеокарта msi GTX 1050 2gb   
44 2018-05-26   7000                           видеокарта GTX  1050 2gb   
45 2018-05-27   2400                         видеокарта GTX  1050 китай   
46 2018-05-28   6000                              geforce GTX  1050 2gb   
47 2018-05-28  10800   видеокарта msi geforce GTX  1050 2g ocv1 2048 mt   
48 2018-05-28  12100   видеокарта msi geforce GTX 1050 gaming x 2g 2...   
49 2018-05-29   2100                         видеокарта GTX  1050 китай   
50 2018-05-31   6500            asus geforce GTX  1050 rog strix oc 2gb   
51 2018-05-31   8500                                          GTX  1050   
52 2018-05-31  10000                           nvidia geforce GTX  1050   
53 2018-06-01   9000               gigabyte geforce GTX  1050 g1 gaming   
54 2018-06-03   7500              новая GTX  1050 2gb (3 года гарантия)   
55 2018-06-03   8000                         GTX  1050 2gb palit stormx   
56 2018-06-03  11990   видеокарта msi geforce GTX  1050 2048mb gaming x   
57 2018-06-05   8000                       новая gigabyte GTX  1050 2gb   
58 2018-06-06   7000                        nvidia palit GTX  1050 2 гб   
59 2018-06-09   7200                  видеокарта msi GTX  1050 2gt ocv1   
60 2018-06-10   7000                                  geforce GTX  1050   
61 2018-06-11   7500                       msi GTX  1050 aero 2g itx oc   
62 2018-06-12   9000                                          GTX  1050   
63 2018-06-14   8000                               видеокарта GTX  1050   
64 2018-06-15   5000                                   продам GTX  1050   
65 2018-06-15   7500                       msi geforce GTX  1050 2gb oc   

                                                 Link ShortName  
41  /sankt-peterburg/tovary_dlya_kompyutera/videok...   GTX1050  
42  /sankt-peterburg/tovary_dlya_kompyutera/gigaby...   GTX1050  
43  /sankt-peterburg/tovary_dlya_kompyutera/videok...   GTX1050  
44  /sankt-peterburg/tovary_dlya_kompyutera/videok...   GTX1050  
45  /sankt-peterburg/tovary_dlya_kompyutera/videok...   GTX1050  
46  /sankt-peterburg/tovary_dlya_kompyutera/geforc...   GTX1050  
47  /sankt-peterburg/tovary_dlya_kompyutera/videok...   GTX1050  
48  /sankt-peterburg/tovary_dlya_kompyutera/videok...   GTX1050  
49  /sankt-peterburg/tovary_dlya_kompyutera/videok...   GTX1050  
50  /sankt-peterburg/tovary_dlya_kompyutera/asus_g...   GTX1050  
51  /sankt-peterburg/tovary_dlya_kompyutera/gtx_10...   GTX1050  
52  /sankt-peterburg/tovary_dlya_kompyutera/nvidia...   GTX1050  
53  /sankt-peterburg/tovary_dlya_kompyutera/gigaby...   GTX1050  
54  /sankt-peterburg/tovary_dlya_kompyutera/novaya...   GTX1050  
55  /sankt-peterburg/tovary_dlya_kompyutera/gtx_10...   GTX1050  
56  /sankt-peterburg/tovary_dlya_kompyutera/videok...   GTX1050  
57  /sankt-peterburg/tovary_dlya_kompyutera/novaya...   GTX1050  
58  /sankt-peterburg/tovary_dlya_kompyutera/nvidia...   GTX1050  
59  /sankt-peterburg/tovary_dlya_kompyutera/videok...   GTX1050  
60  /sankt-peterburg/tovary_dlya_kompyutera/geforc...   GTX1050  
61  /sankt-peterburg/tovary_dlya_kompyutera/msi_gt...   GTX1050  
62  /sankt-peterburg/tovary_dlya_kompyutera/gtx_10...   GTX1050  
63  /sankt-peterburg/tovary_dlya_kompyutera/videok...   GTX1050  
64  /sankt-peterburg_kolpino/tovary_dlya_kompyuter...   GTX1050  
65  /sankt-peterburg/tovary_dlya_kompyutera/msi_ge...   GTX1050

In [31]:

b = GPUnotFarmSearch.loc[64,'Link'] #change index
b = 'https://www.avito.ru' + str(b)
print(b)

https://www.avito.ru/sankt-peterburg_kolpino/tovary_dlya_kompyutera/prodam_gtx_1050_1684467611


In [ ]:
GPUnotFarmSearch.dtypes

In [ ]:
%matplotlib notebook
import matplotlib.pyplot as plt
import seaborn as sns

#sns.set_style("whitegrid")



plt.figure(figsize=(36,12),frameon=False)
plt.box(on=None) 
plt.yticks(np.arange(0, 90001, step=10000))
plt.ylim(0,105000)
plt.grid(color='black', linestyle='-', linewidth=0.1)
plt.legend()

#for items in ModelCounter.items():
#    plt.scatter(x = items[0], y = 80000, alpha=0.3, s=15*int(items[1]), label='GPU amount' + str(int(items[1])))
#    print(items)

#plt.scatter('GTX460',50000)

#plt.scatter([], [], c='k', alpha=0.3, s=area, label=str(area) + ' km$^2$')


ax = sns.stripplot(x='ShortName', y="Price", data = GPUnotFarmFull, jitter=True, size=3)
plt.savefig('AvitoGPUtest', dpi=200)

plt.figure(figsize=(9,6))
sns.violinplot(x='ShortName', y="Price", data=GPUnotFarm, split=True,
               inner="quart")